test/javascripts/turbolinks_test.coffee

```diff
@@ -1,210 +1,946 @@
-        @server.respond()
-
-        assert.equal "Hi there!", document.title
-        assert.equal -1, document.body.textContent.indexOf("YOLO")
-        assert document.body.textContent.indexOf("Hi bob") > 0
-        assert @pushStateStub.notCalled
-        assert @replaceStateStub.notCalled
-
-      it 'calls a user-supplied callback', ->
-        @server.respondWith([200, { "Content-Type": "text/html" }, html_one]);
-
-        your_callback = stub()
-        Turbolinks.visit "/some_request", {partialReplace: true, onlyKeys: ['turbo-area'], callback: your_callback}
-        @server.respond()
-
-        assert your_callback.calledOnce
-
-      it 'script tags are evaluated when they are the subject of a partial replace', ->
-        window.globalStub = stub()
-        @server.respondWith([200, { "Content-Type": "text/html" }, script_response]);
-
-        Turbolinks.visit "/some_request", {partialReplace: true, onlyKeys: ['turbo-area']}
-        @server.respond()
-        assert globalStub.calledOnce
-
-      it 'script tags are not evaluated if they have [data-turbolinks-eval="false"]', ->
-        window.globalStub = stub()
-        @server.respondWith([200, { "Content-Type": "text/html" }, script_response_turbolinks_eval_false]);
-
-        Turbolinks.visit "/some_request", {partialReplace: true, onlyKeys: ['turbo-area']}
-        @server.respond()
-        assert.equal 0, globalStub.callCount
-
-      it 'triggers the page:load event with a list of nodes that are new (freshly replaced)', ->
-        $(document).one 'page:load', (event) ->
-          ev = event.originalEvent
-          assert.equal true, ev.data instanceof Array
-          assert.equal 1, ev.data.length
-          node = ev.data[0]
-
-          assert.equal "turbo-area", node.id
-          assert.equal "turbo-area", node.getAttribute('refresh')
-
-        @server.respondWith([200, { "Content-Type": "text/html" }, html_one]);
-
-        Turbolinks.visit "/some_request", {partialReplace: true, onlyKeys: ['turbo-area']}
-        @server.respond()
-
-        $(document).off 'page:load'
-
-      it 'does not trigger the page:before-partial-replace event more than once', ->
-        handler = stub()
-        $(document).on 'page:before-partial-replace', ->
-          handler()
-          assert handler.calledOnce
-
-        @server.respondWith([200, { "Content-Type": "text/html" }, html_one]);
-
-        Turbolinks.visit "/some_request", {partialReplace: true, onlyKeys: ['turbo-area']}
-        @server.respond()
-
-        $(document).off 'page:before-partial-replace'
-
-      it 'replaces and passes through the outermost nodes if a series of nodes got replaced', ->
-        currentBody = """
-          <div id="div1" refresh="div1">
-            <div id="div2" refresh-always>Refresh-always</div>
-          </div>
-        """
-
-        $("body").append($(currentBody))
-
+        assert.equal('/noScriptsOrLinkInHead', event.originalEvent.data)
```

Comment:
The test needs to return control back to Turbograft for these assertions to be meaningful.  i.e., I can comment out the `preventDefault` and the test still passes.

The assertions should be wrapped by a `setTimeout`, or something.